modules/component-store/src/component-store.ts

```diff
@@ -97,7 +103,14 @@ export class ComponentStore<T extends object> implements OnDestroy {
       this.initState(defaultState);
     }
 
-    this.checkProviderForHooks();
+    // Note that we have still kept the `isDevMode()` condition for cases where
+    // `ngDevMode` would be undefined. Under normal conditions, `ngDevMode` would
+    // always be defined, and it would be evaluated during the compilation into
+    // `if (false)` during the production build, allowing the below expression to
+    // be tree-shaken.
+    if ((typeof ngDevMode === 'undefined' && isDevMode()) || ngDevMode) {
```

Comment:
I'm not sure this makes sense. since `isDevMode()` is `ngDevMode === 'undefined' || !!ngDevMode;`

Reply:
I haven't seen that `isDevMode` implementation was changed to using `ngDevMode`, that's why I used this conditional expression. @markostanimirovic can confirm that using `isDevMode()` is not being stripped out, see comments above.